plugins/dtestconfig.py

```
@@ -0,0 +1,39 @@
+from collections import namedtuple
+
+from nose import plugins
+
+# A class that defines the attributes that have to be defined for configuring
+# a dtest run. namedtuple does what we want -- it's immutable and requires
+# all the attributes to be passed in to be instantiated.
+GlobalConfigObject = namedtuple('GlobalConfigObject', [
+    'vnodes',  # disable or enable vnodes
+])
+
+_CONFIG = None
+
+
+class DtestConfigPlugin(plugins.Plugin):
+    """
+    Pass in configuration options for the dtests.
+    """
+    enabled = True  # if this plugin is loaded at all, we're using it
+    name = 'dtest_config'
+
+    def __init__(self, config=None):
```

Just pushed the markup.

I don't think it's good style to check types. I think the idiomatic thing to do is just let it fail where it'd fail if it doesn't quack like a duck.